### ISSUE TITLE
Event loop is closed

### ISSUE THREAD

**Issue by FarukOzderim** [pytest-dev/pytest-asyncio#371]
While running the tests I am receiving these warning errors, and I suspect this causes CircleCI tests to run forever and not finish.

Reproduction:
Go to the [Commit](https://github.com/hibestil/gradio/tree/2515fede3f0e6739195039443ddbda56db8a84bc) and run
`pytest test/test_utils.py`

Related issues:
#256, #291 

Not sure what causes it and how to solve it, do you have any suggestions @seifertm?

```
======================== 7 passed, 3 warnings in 4.21s ========================

Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2800.0_x64__qbz5n2kfra8p0\lib\asyncio\proactor_events.py", line 116, in __del__
    self.close()
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2800.0_x64__qbz5n2kfra8p0\lib\asyncio\proactor_events.py", line 108, in close
    self._loop.call_soon(self._call_connection_lost, None)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2800.0_x64__qbz5n2kfra8p0\lib\asyncio\base_events.py", line 746, in call_soon
    self._check_closed()
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2800.0_x64__qbz5n2kfra8p0\lib\asyncio\base_events.py", line 510, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
Exception ignored in: <function _ProactorBasePipeTransport.__del__ at 0x000001C46AEF9D30>
Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2800.0_x64__qbz5n2kfra8p0\lib\asyncio\proactor_events.py", line 116, in __del__
    self.close()
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2800.0_x64__qbz5n2kfra8p0\lib\asyncio\proactor_events.py", line 108, in close
    self._loop.call_soon(self._call_connection_lost, None)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2800.0_x64__qbz5n2kfra8p0\lib\asyncio\base_events.py", line 746, in call_soon
    self._check_closed()
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2800.0_x64__qbz5n2kfra8p0\lib\asyncio\base_events.py", line 510, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
Exception ignored in: <function _ProactorBasePipeTransport.__del__ at 0x000001C46AEF9D30>
Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2800.0_x64__qbz5n2kfra8p0\lib\asyncio\proactor_events.py", line 116, in __del__
    self.close()
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2800.0_x64__qbz5n2kfra8p0\lib\asyncio\proactor_events.py", line 108, in close
    self._loop.call_soon(self._call_connection_lost, None)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2800.0_x64__qbz5n2kfra8p0\lib\asyncio\base_events.py", line 746, in call_soon
    self._check_closed()
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2800.0_x64__qbz5n2kfra8p0\lib\asyncio\base_events.py", line 510, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed

```

**Comment by Puh00** [comment]
I stumbled upon a similar issue and solved it by combining these two solutions: [https://github.com/pytest-dev/pytest-asyncio/issues/30#issuecomment-226947196](https://github.com/pytest-dev/pytest-asyncio/issues/30#issuecomment-226947196) and [https://github.com/encode/httpx/issues/914#issuecomment-622586610](https://github.com/encode/httpx/issues/914#issuecomment-622586610)

So the end result is:
```python
# content of conftest.py
import asyncio

@pytest.yield_fixture
def event_loop():
    """Create an instance of the default event loop for each test case."""
    policy = asyncio.WindowsSelectorEventLoopPolicy()
    res = policy.new_event_loop()
    asyncio.set_event_loop(res)
    res._close = res.close
    res.close = lambda: None

    yield res

    res._close()
```

**Comment by FarukOzderim** [comment]
Thx for the suggestion @Puh00, we've solved it with [this](https://github.com/gradio-app/gradio/pull/1595/commits/34ab594d1e6ef2632529c9521c0c751102c67b49#diff-1780a2492dac55d0b4c5ff13cd3cf852a72cdac8995e17b10c87eeb678b3179eR191-R197)

Btw I liked your github page a lot, it rocks 🐱

**Comment by seifertm** [comment]
I was just looking into this issue as you posted your comment.

Glad you could solve it. I assume this can be closed.

**Comment by FarukOzderim** [comment]
So to have it clear, is it a bug related to Python(there are some comments in the threads provided above), or is it related to pytest-asyncio?

**Comment by seifertm** [comment]
@FarukOzderim If in doubt, it's a pytest-asyncio issue.

Honestly, I didn't manage to set up your example and reproduce the error. I checked out the repository, installed the requirements and downloaded >1GB of Python wheels. After that, the test still complained about missing dependencies, which I installed one by one. I stopped after the third and found your comment. No offense intended. I just wanted to point out that a more stripped down example would have been helpful.

I assume it's related to the use of `pytest.fixture(autouse=True)` you already linked (#291)
